docs/userguides/testing.md

```diff
@@ -225,6 +225,102 @@ def test_account_balance(project, owner, receiver, nft):
     assert actual == expect
 ```
 
+## Testing Transaction Failures
+
+Similar to `pytest.raises()`, you can use `ape.reverts()` to assert that contract transactions fail and revert.
+
+From our earlier example we can see this in action:
+
+```python
+def test_authorization(my_contract, owner, not_owner):
+    my_contract.set_owner(sender=owner)
+    assert owner == my_contract.owner()
+
+    with ape.reverts("!authorized"):
+        my_contract.authorized_method(sender=not_owner)
+```
+
+`reverts()` takes two optional parameters:
+
+### `expected_message`
+
+This is the expected revert reason given when the transaction fails.
+If the message in the `ContractLogicError` raised by the transaction failure is empty or does not match the `expected_message`, then `ape.reverts()` will raise an `AssertionError`.
+
+### `dev_message`
+
+This is the expected dev message corresponding to the line in the contract's source code where the error occurred.
+
+Dev messages take the form of a comment in Vyper, and should be placed on the line that may cause a transaction revert:
```

Comment:
Should we include a note about how this is used for gas optimization and keeping revert reasons shorter?

Reply:
Updated in 71d5cc10